src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/ScriptManagerImpl.java

```diff
@@ -73,12 +158,41 @@ public void executeScript(
     }
   }
 
+  private String getUtcTimeStringFromTimestamp(Timestamp timestamp) {
+    Instant instant = timestamp.toInstant();
+    // Replace the "Z" that signifies UTC timezone with "S" that signifies the separator between
+    // seconds and fractional seconds; remove separators that are not necessary for human
+    // interpretation.
+    String instantWithoutNanoseconds =
+        instant
+            .truncatedTo(ChronoUnit.SECONDS)
+            .toString()
+            .replaceAll("[\\-:]", "")
+            .replaceAll("[Z]", "S")
+            .trim();
+    // Keep 6 digits of the fractional seconds from nano, which is equivalent to round(nanos / 10^9
+    // * 10^6) with left-padded zeros.
+    String sixDigitNanos =
+        String.format((Locale) null, "%06d", Math.round(timestamp.getNanos() / Math.pow(10, 3)));
+    return instantWithoutNanoseconds + sixDigitNanos;
+  }
+
   @Override
-  public String getScript(SqlTemplateRenderer sqlTemplateRenderer, String scriptName) {
+  public String getScript(
+      SqlTemplateRenderer sqlTemplateRenderer,
+      String scriptName,
+      ImmutableList<String> sortingColumns) {
     Preconditions.checkArgument(
         scriptsMap.containsKey(scriptName),
         String.format("Script name %s is not available.", scriptName));
-    return sqlTemplateRenderer.renderTemplate(scriptName, scriptsMap.get(scriptName).get());
+    String rawScript = scriptsMap.get(scriptName).get();
+    if (!sortingColumns.isEmpty()) {
+      String orderClause =
+          sortingColumns.stream()
+              .collect(Collectors.joining(", ", "\nORDER BY ", " ASC NULLS FIRST"));
```

Comment:
I think it would be cleaner to provide the order columns as a parameter to the template and have the template create the `ORDER BY` part. This goes against the idea of using templates in the first place.

Reply:
Good idea. Modified accordingly.